nodestream/file_io.py

```diff
@@ -125,3 +127,37 @@ def write_to_file(self, file_path: Path):
                 indent=2,
                 sort_keys=True,
             )
+
+
+# This approach is inspired by https://death.andgravity.com/any-yaml
+#
+# Generally, the idea is that instead of trying to resolve the arguments at the time of parsing the yaml file,
+# we instead create a special object that will resolve the arguments at the time of use.
+#
+# This comes with the side effect that we shift the responsibility of resolving the arguments from the
+# yaml parser to the code that uses the "loaded" yaml data.
+class LazyLoadedArgument:
+    def __init__(self, tag, value):
+        self.tag = tag
+        self.value = value
+
+    def get_value(self):
+        return ArgumentResolver.resolve_arugment_with_alias(self.tag, self.value)
```

Comment:
typo in word `argument`

Reply:
Does this need any error handling?

Reply:
Fixing the typo caused this to be hidden in the diff. My bad. 

re: error handling

Right now, if an exception is thrown (because the tag was wrong or something was thrown trying to resolve it), it would percolate all the way out to the CLI which does a pretty good job of formatting errors and showing where the failure happened. 

If anything, we can catch and raise an exception that has better context (Such as what tag and value it was).